### ISSUE TITLE
Debugger.throw() with PluginDebugOpts.ExceptionHandler error

### ISSUE THREAD

**Issue by Desetude** [Codelanx/CodelanxLib#19]
When Debugger.throw() is called and the logger has a hooked PluginDebugOpts.ExceptionHandler, this exception is thrown:

```java
java.lang.IllegalArgumentException: class java.util.logging.Logger is not provided by class org.bukkit.plugin.java.PluginClassLoader
        at org.bukkit.plugin.java.JavaPlugin.getProvidingPlugin(JavaPlugin.java:525) ~[spigot.jar:git-Spigot-687a019-4c1e934]
        at com.codelanx.codelanxlib.util.ReflectBukkit.getCallingPlugin(ReflectBukkit.java:93) ~[?:?]
        at com.codelanx.codelanxlib.logging.PluginDebugOpts.getPluginOpts(PluginDebugOpts.java:97) ~[?:?]
        at com.codelanx.commons.logging.Debugger$DebugUtil.getOpts(Debugger.java:376) ~[?:?]
        at com.codelanx.codelanxlib.logging.PluginDebugOpts$ExceptionHandler.publish(PluginDebugOpts.java:241) ~[?:?]
        at java.util.logging.Logger.log(Unknown Source) [?:1.8.0_91]
        at org.bukkit.plugin.PluginLogger.log(PluginLogger.java:33) ~[spigot.jar:git-Spigot-687a019-4c1e934]
        at java.util.logging.Logger.doLog(Unknown Source) [?:1.8.0_91]
        at java.util.logging.Logger.log(Unknown Source) [?:1.8.0_91]
        at com.codelanx.commons.logging.Debugger.error(Debugger.java:160) ~[?:?]
```

**Comment by csh** [comment]
>>> When Debugger.throw() is called and the logger has a hooked PluginDebugOpts.ExceptionHandler, this exception is thrown:

```
java.lang.IllegalArgumentException: class java.util.logging.Logger is not provided by class org.bukkit.plugin.java.PluginClassLoader at org.bukkit.plugin.java.JavaPlugin.getProvidingPlugin(JavaPlugin.java:525) ~[spigot.jar:git-Spigot-687a019-4c1e934] at com.codelanx.codelanxlib.util.ReflectBukkit.getCallingPlugin(ReflectBukkit.java:93) ~[?:?] at com.codelanx.codelanxlib.logging.PluginDebugOpts.getPluginOpts(PluginDebugOpts.java:97) ~[?:?] at com.codelanx.commons.logging.Debugger$DebugUtil.getOpts(Debugger.java:376) ~[?:?] at com.codelanx.codelanxlib.logging.PluginDebugOpts$ExceptionHandler.publish(PluginDebugOpts.java:241) ~[?:?] at java.util.logging.Logger.log(Unknown Source) [?:1.8.0_91] at org.bukkit.plugin.PluginLogger.log(PluginLogger.java:33) ~[spigot.jar:git-Spigot-687a019-4c1e934] at java.util.logging.Logger.doLog(Unknown Source) [?:1.8.0_91] at java.util.logging.Logger.log(Unknown Source) [?:1.8.0_91] at com.codelanx.commons.logging.Debugger.error(Debugger.java:160) ~[?:?]
```

FTFY

**Comment by csh** [comment]
```java
java.lang.IllegalArgumentException: class java.util.logging.Logger is not provided by class org.bukkit.plugin.java.PluginClassLoader
       at org.bukkit.plugin.java.JavaPlugin.getProvidingPlugin(JavaPlugin.java:525) ~[spigot.jar:git-Spigot-687a019-4c1e934]
       at com.codelanx.codelanxlib.util.ReflectBukkit.getCallingPlugin(ReflectBukkit.java:93) ~[?:?]
       at com.codelanx.codelanxlib.logging.PluginDebugOpts.getPluginOpts(PluginDebugOpts.java:97) ~[?:?]
       at com.codelanx.commons.logging.Debugger$DebugUtil.getOpts(Debugger.java:376) ~[?:?]
       at com.codelanx.codelanxlib.logging.PluginDebugOpts$ExceptionHandler.publish(PluginDebugOpts.java:241) ~[?:?]
       at java.util.logging.Logger.log(Unknown Source) [?:1.8.0_91]
       at org.bukkit.plugin.PluginLogger.log(PluginLogger.java:33) ~[spigot.jar:git-Spigot-687a019-4c1e934]
       at java.util.logging.Logger.doLog(Unknown Source) [?:1.8.0_91]
       at java.util.logging.Logger.log(Unknown Source) [?:1.8.0_91]
       at com.codelanx.commons.logging.Debugger.error(Debugger.java:160) ~[?:?]
```

FTFY

**Comment by 1Rogue** [comment]
Are you compiled for the latest revisions? I've just updated the maven repo, as the line numbers do not match current code.

If you still get the error, post it again in this issue report.

**Comment by Desetude** [comment]
Compiled using the latest version on GitHub, same error, different line numbers.

```java
java.lang.IllegalArgumentException: class java.util.logging.Logger is not provided by class org.bukkit.plugin.java.PluginClassLoader
	at org.bukkit.plugin.java.JavaPlugin.getProvidingPlugin(JavaPlugin.java:525) ~[spigot.jar:git-Spigot-687a019-4c1e934]
	at com.codelanx.codelanxlib.util.ReflectBukkit.getCallingPlugin(ReflectBukkit.java:97) ~[?:?]
	at com.codelanx.codelanxlib.logging.PluginDebugOpts.getPluginOpts(PluginDebugOpts.java:273) ~[?:?]
	at com.codelanx.commons.logging.Debugger$DebugUtil.getOpts(Debugger.java:390) ~[?:?]
	at com.codelanx.codelanxlib.logging.PluginDebugOpts$ExceptionHandler.publish(PluginDebugOpts.java:228) ~[?:?]
	at java.util.logging.Logger.log(Unknown Source) [?:1.8.0_91]
	at org.bukkit.plugin.PluginLogger.log(PluginLogger.java:33) ~[spigot.jar:git-Spigot-687a019-4c1e934]
	at java.util.logging.Logger.doLog(Unknown Source) [?:1.8.0_91]
	at java.util.logging.Logger.log(Unknown Source) [?:1.8.0_91]
	at com.codelanx.commons.logging.Debugger.error(Debugger.java:164) ~[?:?]
```

**Comment by 1Rogue** [comment]
I see the issue, it's a design implication so you'll need to give me a moment to actually decide how to go about fixing it.

**Comment by 1Rogue** [comment]
Fixed in version `0.3.1-SNAPSHOT`.
